### ISSUE TITLE
renderPagination seems to run ~8 times on intial render

### ISSUE THREAD

**Issue by RCRJ** [sag1v/react-elastic-carousel#58]
I'm wondering if this is normal, I created a pretty bare-bones example in codesandbox were I am rendering a Material UI Pagination component from the renderPagination function.

I also tried memoizing the component however it seems like the pagination is always rendered multiple times.

[![Edit interesting-lamarr-yszok](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/interesting-lamarr-yszok?fontsize=14&hidenavigation=1&theme=dark)

**Comment by sag1v** [comment]
Thanks @RCRJ.

TL;DR: There is abug, we are passing the `pages` array as a new array on each render.

This get rendered multiple times because the root `Carousel` gets state-updates which trigger a re-render (state update due to width / height changes from initial paint).

Also keep in mind the React will render twice everything below `<React.StrictMode>`.

With that said, `react-elastic-carousel` will break your caching because of [one single line](https://github.com/sag1v/react-elastic-carousel/blob/master/src/react-elastic-carousel/components/Carousel.js#L540). It passes a new array of `pages` on each render thus breaking the default memoization check e.g: `prevProps !== nextProps`, or in our specific case -  `prevProps.pages !== nextProps.pages`

We can confirm that if we just remove `pages` from the dependencies array, although this "work" as expected memo wise, we have a broken pagination.

So bottom line, we have a bug and i suspect that the solution would be just passing the same array or maybe caching it.

I'm open to PR's if you want to take it :slightly_smiling_face:

**Comment by sag1v** [comment]
fixed in `v0.6.7`
